docker/package/package_generator.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python3
+# SPDX-FileCopyrightText: 2020 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+import os, sys, subprocess, json
+from distutils.dir_util import copy_tree
+
+class Package:
+    def __init__(self, name, desc):
+        self.name = name
+        self.desc = desc
+
+if sys.argv[1] == "source":
+    is_source = True
+elif sys.argv[1] == "binary":
+    is_source = False
+else:
+    raise Exception("Unexpected package format, only 'source' and 'binary' are supported.")
+
+package_to_build = None
+if len(sys.argv) == 3:
+    package_to_build = sys.argv[2]
+
+date = subprocess.check_output(["date", "-R"]).decode().strip()
+meta = json.load(open("../meta.json", "r"))
+tezos_sources = json.load(open("../nix/nix/sources.json", "r"))["tezos"]
+
+common_deps = ["libev-dev", "libgmp-dev", "libhidapi-dev", "m4", "perl", "pkg-config",
+               "wget", "unzip", "rsync"]
+
+active_protocols = json.load(open("../protocols.json", "r"))["active"]
+
+version = f"{tezos_sources['ref'][1:]}"
+release = f"{meta['release']}"
+package_version = f"{meta['epoch']}:0ubuntu{version}-{release}"
+
+
+def gen_control_file(pkg: Package, out):
+    str_deps = ", ".join(common_deps)
+    file_contents = f'''
+Source: {pkg.name.lower()}
+Section: utils
+Priority: optional
+Maintainer: {meta['maintainer']}
+Build-Depends: debhelper (>=9), autotools-dev, {str_deps}
+Standards-Version: 3.9.6
+Homepage: https://gitlab.com/tezos/tezos/
+
+Package: {pkg.name.lower()}
+Architecture: amd64
+Depends: ${{shlibs:Depends}}, ${{misc:Depends}}
+Description: {pkg.desc}
+'''
+    with open(out, 'w') as f:
+        f.write(file_contents)
+
+def gen_makefile(pkg: Package, out):
+    makefile_contents =f'''
+.PHONY: install
+
+BINDIR=/usr/bin
+
+{pkg.name}:
+	./compile.sh
+	cp $(CURDIR)/opam/default/bin/{pkg.name} {pkg.name}
+
+install: {pkg.name}
+	mkdir -p $(DESTDIR)$(BINDIR)
+	cp $(CURDIR)/{pkg.name} $(DESTDIR)$(BINDIR)
+'''
+    with open(out, 'w') as f:
+        f.write(makefile_contents)
+
+def gen_changelog(pkg: Package, ubuntu_version, maintainer, date, out):
+    changelog_contents = f'''{pkg.name.lower()} ({package_version}) {ubuntu_version}; urgency=medium
+
+  * Publish {version}-{release} version of {pkg.name}
+
+ -- {maintainer} {date}'''
+    with open(out, 'w') as f:
+        f.write(changelog_contents)
+
+def gen_rules(out):
+    rules_contents = '''#!/usr/bin/make -f
+
+%:
+	dh $@ '''
+    with open(out, 'w') as f:
+        f.write(rules_contents)
+
+packages = [
+    Package("tezos-client",
+            "CLI client for interacting with tezos blockchain"),
+    Package("tezos-admin-client",
+            "Administration tool for the node") ,
+    Package("tezos-node",
+            "Entry point for initializing, configuring and running a Tezos node"),
+    Package("tezos-signer",
+            "A client to remotely sign operations or blocks")
+]
+
+daemons = ["baker", "accuser", "endorser"]
```

Comment:
AFAIU node is also a daemon, isn't it? But apparently it doesn't need protocol suffix, unlike other daemons. So wouldn't it be more appropriate to call this list somehow differently? :)